core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -117,112 +131,90 @@ public TenantObject setEnabled(final boolean flag) {
     }
 
     /**
-     * Gets the subject DN of this tenant's configured trusted
-     * certificate authority.
+     * Get the subject DN of the configured trusted CA.
      * 
      * @return The DN or {@code null} if no CA has been set.
      */
     @JsonIgnore
+    @Deprecated
     public X500Principal getTrustedCaSubjectDn() {
 
-        final JsonObject trustedCa = getProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA, JsonObject.class);
-        if (trustedCa == null) {
-            return null;
+        final Set<X500Principal> subjectNames = getTrustedCaSubjectDns();
+        if (subjectNames == null) {
+           return null;
         } else {
-            return Optional
-                    .ofNullable(getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, String.class))
-                    .map(dn -> new X500Principal(dn)).orElse(null);
+            return subjectNames.stream().findFirst().orElse(null);
         }
     }
 
     /**
-     * Sets the trusted certificate authority to use for authenticating
+     * Adds the trusted certificate authority to use for authenticating
      * devices of this tenant.
      * 
-     * @param publicKey The CA's public key.
-     * @param subjectDn The CA's subject DN.
+     * @param publicKey The trusted CA's public key.
+     * @param subjectDn The trusted CA's subject DN.
+     * @param notBefore The start date in which the trustedCa becomes valid.
+     * @param notAfter The end date in which the trustedCa becomes invalid.
+     * 
      * @return This tenant for command chaining.
      * @throws NullPointerException if any of the parameters is {@code null}.
      */
     @JsonIgnore
-    public TenantObject setTrustAnchor(final PublicKey publicKey, final X500Principal subjectDn) {
+    public TenantObject addTrustAnchor(final PublicKey publicKey, final X500Principal subjectDn,
+            final Instant notBefore, final Instant notAfter) {
 
         Objects.requireNonNull(publicKey);
         Objects.requireNonNull(subjectDn);
 
         final JsonObject trustedCa = new JsonObject();
         trustedCa.put(TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, subjectDn.getName(X500Principal.RFC2253));
         trustedCa.put(TenantConstants.FIELD_PAYLOAD_PUBLIC_KEY, publicKey.getEncoded());
+        trustedCa.put(TenantConstants.FIELD_PAYLOAD_NOT_BEFORE, notBefore.toString());
```

Comment:
I would rather use *toEpochMillis()* here which uses less memory and also does not require special formatting.
BTW using a class' *toString()* method should not generally be expected to yield a specific defined format which you can rely on containing certain information at specific places. Instant.toString() states that it uses an ISO format, but this might easily change in the next Java version ...

Reply:
we seem to be back to the old problem ;-)
IMHO we should explicitly use a `DateTimeFormatter` to produce the intended ISO format instead of relying on *toString()*.
The same holds true for parsing the values ...